src/core/execute.c

```diff
@@ -5950,6 +6051,19 @@ int exec_context_get_clean_directories(
                                         return r;
                         }
                 }
+
+                char **d;
+                STRV_FOREACH_PAIR(i, d, c->directories[t].symlinks) {
+                        char *j;
+
+                        j = path_join(prefix[t], *d);
+                        if (!j)
+                                return -ENOMEM;
+
+                        r = strv_consume(&l, j);
+                        if (r < 0)
+                                return r;
+                }
```

Comment:
this loop appears in a very similar fashion too. maybe add common helper, too?  path_join_strv() or so?

Reply:
wouldn't quite work, as here we iterate on tuples (and only want to pick one), above on single directories